### PR TITLE
SAF 52 - Toggle lend mode

### PR DESCRIPTION
**Pull request by matzapata**
## Overview

Jira tickets: [SAF-52](https://wakeuplabs.atlassian.net/browse/SAF-52)

Switch borrow position between stable and variable rate mode

## Demo

https://github.com/user-attachments/assets/0f4941c0-733c-4448-b8aa-9c03b7079109

## Extra notes

Based of [#30](https://github.com/wakeuplabs-io/sovryn-dapp/pull/30) Pointing here to make the PR reviewable in the meantime
There's a bug [SAF-55](https://wakeuplabs.atlassian.net/browse/SAF-55) so I can't test for ETH. I'll fix that in a separate pr and recheck here.

### CODE REVIEW

**Review by luchopcerra — APPROVED** [review]
LGTM 🚀

File: apps/frontend/src/hooks/aave/useAaveBorrow.tsx [hunk 1]
```diff
@@ -55,5 +55,28 @@ export const useAaveBorrow = () => {
...
+      asset: AssetDetailsData,
+      currentRateMode: BorrowRateMode,
+      opts?: TransactionFactoryOptions,
+    ) => {
+      if (!aaveBorrowTransactionsFactory) {
+        return;
+      }
+
+      setTransactions(
+        await aaveBorrowTransactionsFactory.swapBorrowRateMode(
```
Comment (njaume) [inline apps/frontend/src/hooks/aave/useAaveBorrow.tsx:69]: how are you handling errors here ?

**Review by njaume — APPROVED** [review]
LGTM! just a question

File: apps/frontend/src/hooks/aave/useAaveBorrow.tsx (hunk 1 above)
Comment (matzapata) [reply apps/frontend/src/hooks/aave/useAaveBorrow.tsx:69]: Actually it's a good point, I'm not doing any. There shouldn't be any errors arising from user behavior. But I'll add these checks in https://github.com/wakeuplabs-io/sovryn-dapp/pull/33

**Review by juandahl — COMMENTED** [review]
LGTM

**Pull request merged**